### PR TITLE
[terra-application-v2] Terra application docs

### LINKED ISSUE #199: [terra-application-v2] Create terra-application-docs package
# Story

## Description

As a terra developer I want to create a terra-application-docs package

## Motivation

So that we have a location to store terra-application documentation

## Definition of Done

We'll know we're done when the terra-application-docs package has been merged to the v2 branch.

### PR DESCRIPTION
**Pull request by mjhenkes**
### Summary

Sets up the doc repo for terra-application and moves the terra-dev-site folder from terra-application and terra-dev-site over to it.

Closes #199

### Deployment Link

https://terra-applic-.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-dev-site/src/webpack/plugin/applyDefaults.js [hunk 1]
```diff
@@ -53,6 +53,7 @@ const applyDefaults = ({
   enableDebugLogging = false,
   additionalConfig = {},
   excludeChunks = [],
+  contentDirectory = 'terra-dev-site',
```
Comment (mjhenkes) [inline packages/terra-dev-site/src/webpack/plugin/applyDefaults.js:56]: I modified this to allow consumers to change the root content directory for their plug in via config. Previously you had to make a new 'plugin' to modify this. I used this to break out the terra-dev-site test site into it's own folder.

File: packages/terra-application-docs/src/terra-dev-site/app/components.2/example/CustomBannerVariant.module.scss [hunk 2]
```diff
@@ -1,7 +1,8 @@
-@import '~terra-mixins/lib/Mixins';
+@import 'clinical-lowlight-theme/CustomBannerVariant.module';
+@import 'orion-fusion-theme/CustomBannerVariant.module';
 
 :local {
-  .clinical-lowlight-theme {
-    @include terra-inline-svg-var('--terra-application-notification-banner-doc-custom-variant-icon-background', "<svg viewBox='0 0 49 49' xmlns='http://www.w3.org/2000/svg'><path fill='#654A08' d='M24 2.3L45.7 24 24 45.7 2.3 24 24 2.3M24 0c-.3 0-.5.1-.8.3L.3 23.2c-.4.5-.4 1.1 0 1.6l22.9 22.9c.2.2.5.3.8.3s.5-.1.8-.3l22.9-22.9c.4-.5.4-1.1 0-1.6L24.8.3c-.3-.2-.5-.3-.8-.3zm0 8c3.9 3.9 4 13 4 13H15v6c0 10 9 13 9 13-3.9-3.9-4-13-4-13h13v-6c0-10-9-13-9-13z' ></path></svg>");
+  .custom-notification-banner-icon {
```
Comment (yuderekyu) [inline packages/terra-application-docs/src/terra-dev-site/app/components.2/example/CustomBannerVariant.module.scss:5]: 👍🏼

File: webpack.config.js [hunk 3]
```diff
@@ -79,19 +80,19 @@ const terraApplicationConfig = (env = {}) => ({
       titleConfig: {
         title: 'Terra Dev Site - Extended',
       },
-      additionalSearchDirectories: [
-        path.resolve(process.cwd(), 'node_modules', 'terra-list', 'lib', 'terra-dev-site'),
-      ],
+      // additionalSearchDirectories: [
```
Comment (yuderekyu) [inline webpack.config.js]: remove?

**Review by yuderekyu — APPROVED** [review]

File: webpack.config.js (hunk 3 above)
Comment (mjhenkes) [reply webpack.config.js]: It's kind of a reminder. I need to backfill exercising this part of the API with something else. This is a victim of the new strategy of breaking dev sites out from individual packages and into docs packages per mono repo. Eventually terra-list won't have anything to find.

Comment (yuderekyu) [reply webpack.config.js]: not a big deal, but a todo comment in the interim would be helpful

Comment (mjhenkes) [reply webpack.config.js]: I went ahead and took care of the todo :)

**Review by dkasper-was-taken — APPROVED** [review]

**Pull request merged**